### PR TITLE
Fixed #27505 -- Allowed customizing Paginator's error messages.

### PR DESCRIPTION
**Pull request by mgaligniana**
Hi!

This PR fixes [#27505](https://code.djangoproject.com/ticket/27505) applying the [Tim's comment](https://github.com/django/django/pull/7578#issuecomment-264017771).

To run the specific test please do `./runtests.py pagination.tests.PaginationTests.test_custom_paginator_exception_messages`

Thank you!

### CODE REVIEW

File: django/core/paginator.py [hunk 1]
```diff
@@ -28,13 +28,30 @@ class Paginator:
...
+    ):
         self.object_list = object_list
         self._check_object_list_is_ordered()
         self.per_page = int(per_page)
         self.orphans = int(orphans)
         self.allow_empty_first_page = allow_empty_first_page
+        self.error_messages = (
+            self.default_error_messages
+            if error_messages is None
+            else self.default_error_messages | error_messages
```
Comment (mgaligniana) [inline django/core/paginator.py:53]: I just learnt `dict_1 | dict_2` merges two dicts. I can use `.copy` and `.update` if it's more friendly|readeable

File: docs/ref/paginator.txt [hunk 2]
```diff
@@ -56,6 +56,31 @@ For examples, see the :doc:`Pagination topic guide </topics/pagination>`.
     ``False`` and ``object_list`` is  empty, then an ``EmptyPage`` error will
     be raised.
 
+.. attribute:: Paginator.error_messages
```
Comment (mgaligniana) [inline docs/ref/paginator.txt:59]: Almost the same documentation as https://docs.djangoproject.com/en/4.1/ref/forms/fields/#error-messages

**Comment by mgaligniana** [comment]
I couldn't fix the `blackend-docs` step, sorry :(

File: django/core/paginator.py [hunk 3]
```diff
@@ -28,13 +28,30 @@ class Paginator:
     # Translators: String used to replace omitted page numbers in elided page
     # range generated by paginators, e.g. [1, 2, '…', 5, 6, 7, '…', 9, 10].
     ELLIPSIS = _("…")
-
-    def __init__(self, object_list, per_page, orphans=0, allow_empty_first_page=True):
+    default_error_messages = {
+        "page_not_an_integer": _("That page number is not an integer"),
+        "page_not_a_positive_integer": _("That page number is less than 1"),
+        "no_results": _("That page contains no results"),
+    }
```
Comment (felixxm) [inline django/core/paginator.py:35]: I'd use shorter keys, e.g.
```suggestion
    default_error_messages = {
        "invalid_page": _("That page number is not an integer"),
        "min_page": _("That page number is less than 1"),
        "no_results": _("That page contains no results"),
    }
```

File: docs/ref/paginator.txt (hunk 2 above)
Comment (felixxm) [reply docs/ref/paginator.txt:59]: `versionadded:: 5.0` annotation is missing. I'd also document available keys and add release notes.

File: tests/pagination/tests.py [hunk 4]
```diff
@@ -128,6 +128,24 @@ def test_invalid_page_number(self):
...
+    def test_custom_paginator_exception_messages(self):
+        custom_messages = {
+            "page_not_an_integer": "Ese número de página no es un número entero",
+            "page_not_a_positive_integer": "Ese número de página es inferior a 1",
+            # "no_results" is not defined, so default will be shown.
+        }
+        paginator = Paginator([1, 2, 3], 2, error_messages=custom_messages)
+        with self.assertRaisesMessage(
+            PageNotAnInteger, "Ese número de página no es un número entero"
+        ):
```
Comment (felixxm) [inline tests/pagination/tests.py:140]:
```suggestion
        msg = "..."
        with self.assertRaisesMessage(PageNotAnInteger, msg):
```

**Review by felixxm — COMMENTED** [review]
@mgaligniana Thanks :+1:

File: django/core/paginator.py (hunk 3 above)
Comment (mgaligniana) [reply django/core/paginator.py]: Great! Change applied!

File: docs/ref/paginator.txt (hunk 2 above)
Comment (mgaligniana) [reply docs/ref/paginator.txt:59]: I always forget the version tag 🤦 . Thank you!

This is how it looks like:

image

**Review by felixxm — APPROVED** [review]
@mgaligniana Thanks :+1: 

I pushed small edits to docs.

**Pull request merged**